modules/core/server/controllers/core.routes.controller.js

```diff
@@ -241,6 +241,30 @@ var setModel = function (Dbclass, name) {
 	};
 };
 exports.setModel = setModel;
+
+var setModelByQuery = function () {
+	return function (req, res, next) {
+		var name = req.query.collection;
+		var Dbclass;
+		var fields;
+		switch(name) {
```

Comment:
Declare Dbclass at the top.
Remove the switch statement, turn it into an IF.
Reject if the name != 'documents'